corehq/apps/hqadmin/views/users.py

```diff
@@ -89,10 +93,15 @@ def post(self, request, *args, **kwargs):
             request.user.backend = 'django.contrib.auth.backends.ModelBackend'
             login(request, request.user)
             return HttpResponseRedirect('/')
+        all_errors = form.errors.pop('__all__', None)
+        if all_errors:
+            messages.error(request, ','.join(all_errors))
+        if form.errors:
+            messages.error(request, form.errors)
```

Comment:
Could you instead render the form errors in the html?  Django has good support for this kind of thing:
https://docs.djangoproject.com/en/1.11/topics/forms/#rendering-form-error-messages

I expect crispy_forms does too - doesn't it automatically render both field specific errors and `non_field_errors`?

Reply:
Interesting... i tried rendering errors with fields but that didn't work. Going to not worry about it for this and leave it the way it is. Sorry.